rosidl_generator_c/rosidl_generator_c/__init__.py

```diff
@@ -209,9 +220,13 @@ def basic_value_to_c(type_, value):
         return f'{value}ull'
 
     if 'float' == type_.typename:
+        if 'nan' == value:
```

Comment:
Can we use `isnan(float(value))` instead? that would be more reliable check for it?

Reply:
Yes, that makes sense. That way it would catch any `Nan` or `NAN` or similar :+1: